### PR TITLE
pipelines resource feature doc update

### PR DESCRIPTION
**Pull request by ashokirla**
This doc update is for the pipeline resources we have introduced recently into YAML pipelines. 
https://docs.microsoft.com/en-us/azure/devops/release-notes/2019/sprint-159-update#cd-capabilities-for-your-multi-stage-yaml-pipelines

### CODE REVIEW

**Comment by PRMerger19** [comment]
@ashokirla : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/pipelines/yaml-schema.md [hunk 1]
```diff
@@ -875,6 +876,83 @@ steps:
 ::: moniker-end
 
 ## Resources
+Any external service that is consumed as part of your pipeline is a resource. An example of a resource can be another CI/CD pipeline that produces artifacts (say Azure pipelines, Jenkins etc.), code repositories (GitHub, Azure Repos, Git), container image registries (ACR, Docker hub etc.).
+
+Resources in YAML represent sources of types pipelines, repositories and containers.
+
+#### Schema
```
Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: Incorrect heading, should be `###`

Comment (vtbassmatt) [reply docs/pipelines/yaml-schema.md]: I'd also probably call this "General schema" or something, to differentiate it from the explicit schemas you'll show below.

File: docs/pipelines/yaml-schema.md [hunk 2]
```diff
@@ -875,6 +876,83 @@ steps:
...
+
+```yaml
+resources:
+  pipelines: [ pipeline ]  
+  repositories: [ repository ]
+  containers: [ container ]
+```
+
+### Pipeline resource
+If you have an Azure Pipeline that produces artifacts, you can consume the artifacts by defining a `pipelines` resource. And you can also enable pipeline completion triggers.
```
Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: Please put a newline at the end of each sentence. Makes it easier to maintain the doc over time.

Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: Nit: shouldn't this be "by defining a `pipeline` resource" (note singular)?

File: docs/pipelines/yaml-schema.md [hunk 3]
```diff
@@ -875,6 +876,83 @@ steps:
...
+    source: MyCIPipelineA
+  - pipeline: MyAppB
+    source: MyCIPipelineB
+    trigger: true
+  - pipeline: MyAppC
+    project:  DevOpsProject 
+    source: MyCIPipelineC
+    branch: releases/M159  
+    version: 20190718.2 
+    trigger:     .
```
Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: Typo: extraneous `.`

File: docs/pipelines/yaml-schema.md [hunk 4]
```diff
@@ -875,6 +876,83 @@ steps:
...
+        include:  
+        - master
+        - releases/*
+        exclude:   
+        - users/*  
+```
+>[!IMPORTANT]
+>When you define the resource trigger, if the `pipeline` resource is from the same repo as the current pipeline, we will follow the same branch and commit on which the event is raised. However, if the `pipeline` resource is from a different repo, the current pipeline is triggered on the master branch.
+
+#### `pipeline` resource meta-data as pre-defined variables.
```
Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: I'm not convinced this belongs in the schema doc. It's definitely worth putting in another topic and linking to from here, though.

File: docs/pipelines/yaml-schema.md [hunk 5]
```diff
@@ -875,6 +876,83 @@ steps:
...
+resources.pipeline.<Alias>.runID
+resources.pipeline.<Alias>.runURI
+resources.pipeline.<Alias>.sourceBranch 
+resources.pipeline.<Alias>.sourceCommit
+resources.pipeline.<Alias>.sourceProvider 
+resources.pipeline.<Alias>.requestedFor
+resources.pipeline.<Alias>.requestedForID
+```
+
+You can consume artifacts from pipeline resource using `-download` task. See the [download](yaml-schema.md#download) keyword.
```
Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: Typo: don't need the `-` before `download`.

File: docs/pipelines/yaml-schema.md [hunk 6]
```diff
@@ -1598,25 +1676,33 @@ Learn more about [publishing artifacts](./artifacts/pipeline-artifacts.md#publis
 
 ## Download
 
-`download` is a shortcut for the [Download Pipeline Artifact task](tasks/utility/download-pipeline-artifact.md). It will download one or more artifacts associated with the current run to `$(Pipeline.Workspace)`. It can also be used to disable automatic downloading of artifacts in classic release and deployment jobs.
+`download` is a shortcut for the [Download Pipeline Artifact task](tasks/utility/download-pipeline-artifact.md). It will download one or more artifacts associated with the current run or from another azure pipeline that is associated as a `pipeline` resource.
```
Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: Branding: "from another Azure pipeline".

File: docs/pipelines/yaml-schema.md [hunk 7]
```diff
@@ -1598,25 +1676,33 @@ Learn more about [publishing artifacts](./artifacts/pipeline-artifacts.md#publis
...
   patterns: '**/.js'
+- download: MyAppA   # downloads artifacts available as part of the pipeline resource
+
 ```
+### Artifact download location
+Artifacts from current pipeline are downloaded to `$(Pipeline.Workspace)/`. 
+Artifacts from the associated `pipeline` resource are downloaded to `$(Pipeline.Workspace)/<pipeline resource identifier>/`.
+
+### Automatic download in deployment jobs
+All the available artifacts from current pipeline as well as from associated pipeline resources are automatically downloaded in deployment jobs and made available for your deployment. However you can choose to not download by specifiying `- download: none`
```
Comment (vtbassmatt) [inline docs/pipelines/yaml-schema.md]: Nit and spelling, should be: "by specifying \`download: none`"

File: docs/pipelines/yaml-schema.md (hunk 1 above)
Comment (ashokirla) [reply docs/pipelines/yaml-schema.md]: Done

File: docs/pipelines/yaml-schema.md (hunk 4 above)
Comment (ashokirla) [reply docs/pipelines/yaml-schema.md]: Sure. Today the resources doc (vsts-docs/docs/pipelines/process/resources.md) needs to be redone. We need to move variable groups, service connections and agent pools out of it and create a detailed concept doc about resources that talks about pipelines, repos, containers, builds and packages. As part of it I will move this section. Until then I am thinking of keeping this info here.

**Comment by PRMerger20** [comment]
@ashokirla : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by vtbassmatt** [comment]
#sign-off

**Comment by ktoliver** [comment]
Merging the PR.

I'm requesting an edit pass for this article (in the private repository). An editor will contact the listed author soon after these updates are published. Thanks.

**Pull request merged**